scraper/src/mindtouch2zim/context.py

```diff
@@ -0,0 +1,144 @@
+import argparse
+import logging
+import os
+import re
+import threading
+from pathlib import Path
+
+import requests
+from zimscraperlib.__about__ import __version__ as SCRAPERLIB_VERSION  # noqa: N812
+from zimscraperlib.constants import NAME as SCRAPERLIB_NAME
+from zimscraperlib.download import get_session
+
+from mindtouch2zim.constants import (
+    NAME,
+    STANDARD_KNOWN_BAD_ASSETS_REGEX,
+    VERSION,
+    logger,
+)
+
+
+class Context:
+    """Class holding every contextual / configuration bits which can be moved
+
+    Used to easily pass information around in the scraper. One singleton instance is
+    always available.
+    """
+
+    # info passed in User-Agent header of web requests
+    contact_info: str = "https://www.kiwix.org"
+
+    # web session to use everywhere
+    web_session: requests.Session = get_session()
+
+    # temporary folder to store temporary assets (e.g. cached API response)
+    tmp_folder: Path
+
+    # temporary folder to store cached API response
+    cache_folder: Path
+
+    # folder where the ZIM will be built
+    output_folder: Path = Path(os.getenv("MINDTOUCH_OUTPUT", "/output"))
+
+    # folder where Vue.JS UI has been built
+    zimui_dist: Path = Path(os.getenv("MINDTOUCH_ZIMUI_DIST", "../zimui/dist"))
+
+    # Path to store the progress JSON file to
+    stats_filename: Path | None = None
+
+    # URL to illustration to use for ZIM illustration and favicon
+    illustration_url: str | None = None
+
+    # Do not fail if ZIM already exists, overwrite it
+    overwrite_existing_zim: bool = False
+
+    # number of assets processed in parallel
+    assets_workers: int = 10
+
+    # known bad assets
+    bad_assets_regex: re.Pattern | None = re.compile(STANDARD_KNOWN_BAD_ASSETS_REGEX)
+
+    # maximum amount of bad assets
+    bad_assets_threshold: int = 10
+
+    # current processing info to use for debug message / exception
+    _processing_step: threading.local = threading.local()
```

Comment:
I think the name of this (maybe not this holder but the API) should make it very clear that this is local-only of thread-local. Otherwise that would surely turn a future maintainer crazy.